### PR TITLE
Reduce spacing for optimize setup.

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue #4028 

## Relevant technical choices
The styling is pretty specific to the Optimize setup form since we have `flex-wrap: wrap` on the flex items. This means we don't have much control over when the elements are stacked in order to remove the extra margin.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/sass/components/setup/_googlesitekit-setup.scss [hunk 1]
```diff
@@ -114,6 +114,26 @@
 		padding-left: 10px;
 		padding-right: 10px;
 	}
+
+	.googlesitekit-optimize-setup__form {
```
Comment (aaemnnosttv) [inline assets/sass/components/setup/_googlesitekit-setup.scss:118]: I think these styles may not actually be necessary. Most of the styling appears to be fixed by wrapping the optimize field with a `div`.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @asvinb !

As per my last comment below, I found that the main problem here seems to be the missing wrapping div for the `OptimizeIDField`. All other similar components `AdsConversionIDTextField`, `ProfileNameTextField`, `WebContainerNameTextField`, etc all have wrapping divs (some with classes) but for some reason `OptimizeIDField` uses a `Fragment` which is what causes the `googlesitekit-setup-module__inputs > div` style to add too much margin where it shouldn't.

Let's add this missing wrapper and then only add any additional styles as needed.

File: assets/js/modules/optimize/components/common/OptimizeIDField.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ export default function OptimizeIDField() {
 				name="optimizeID"
 				onChange={ onChange }
 				helperText={
-					<HelperText>
+					<HelperText persistent>
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/modules/optimize/components/common/OptimizeIDField.js:64]: 👍

File: assets/sass/components/setup/_googlesitekit-setup-module.scss [hunk 3]
```diff
@@ -128,7 +147,11 @@
 		}
 
 		+ .googlesitekit-error-text {
-			margin-top: -24px;
```
Comment (aaemnnosttv) [inline assets/sass/components/setup/_googlesitekit-setup-module.scss:131]: 👍

**Review by aaemnnosttv — APPROVED** [review]
LGTM 👍  Thanks @asvinb !

**Pull request merged**